std/binary.asm

```diff
@@ -15,15 +16,20 @@ machine Binary(latch, operation_id) {
     col fixed latch(i) { if (i % 4) == 3 { 1 } else { 0 } };
     col fixed FACTOR(i) { 1 << (((i + 1) % 4) * 8) };
 
-    col fixed P_A(i) { i % 256 };
-    col fixed P_B(i) { (i >> 8) % 256 };
-    col fixed P_operation(i) { (i / (256 * 256)) % 3 };
+    // TOOD would be nice with destructuring assignment for arrays.
+    let inputs: (int -> int)[] = cross_product([3, 256, 256]);
+    let a = inputs[2];
+    let b = inputs[1];
+    let op = inputs[0];
+    col fixed P_A(i) { a(i) };
+    col fixed P_B(i) { b(i) };
+    col fixed P_operation(i) { op(i)};
```

Comment:
They are used in the lookup below. Also, a and b are `int -> int` functions (we need that because if the binary operators) and the columns have the implicit conversion to fe.

Reply:
Or, another idea, `cross_product()` could also receive a `(int[] -> int)` to populate the result. But that assumes that there is only one output column.